engine.io-server/src/main/java/io/socket/engineio/server/EngineIoServer.java

```diff
@@ -23,7 +24,10 @@
 @SuppressWarnings("WeakerAccess")
 public final class EngineIoServer extends Emitter {
 
-    private final Map<String, EngineIoSocket> mClients = new TreeMap<>();
+    private final Map<String, EngineIoSocket> mClients = Collections.synchronizedSortedMap(new TreeMap<>());
```

Comment:
Using synchronization is probably a bad idea when there's a lot of incoming requests.
Would it be possible to use a ConcurrentHashMap here?

Reply:
I did think of it but was going for minimal changes to the existing implementation. I'll do this.